### PR TITLE
Log app ID as SYSTEM/INFO during initialization

### PR DESCRIPTION
**Pull request by smcv**
This is not shown by default, but will be shown when using SDL_LOGGING=system=info or higher, where it will hopefully nudge app authors towards initializing this information.

### CODE REVIEW

**Comment by smcv** [comment]
Should we perhaps log the `SDL_GetRevision()` here, too?

**Comment by slouken** [comment]
I don't think it'll nudge anyone, but it might be helpful. I think adding the SDL version and revision is a good idea.

File: src/SDL.c [hunk 1]
```diff
@@ -251,10 +251,27 @@ void SDL_SetMainReady(void)
...
     SDL_InitTLSData();
     SDL_InitEnvironment();
     SDL_InitTicks();
     SDL_InitFilesystem();
+
+    if (!done_info) {
+        const char *value;
+
+        SDL_LogInfo(SDL_LOG_CATEGORY_SYSTEM, "SDL %d.%d.%d, revision: %s",
+                    SDL_MAJOR_VERSION, SDL_MINOR_VERSION, SDL_MICRO_VERSION, SDL_REVISION);
```
Comment (smcv) [inline src/SDL.c:265]: In practice this should typically log something like:

```
SDL 3.1.3, revision SDL-3.1.3 (Debian 3.1.3+ds-1)
```

or maybe

```
SDL 3.1.2, revision SDL-prerelease-3.1.2-1181-gc5a99f151
```

Suitable?

Comment (slouken) [reply src/SDL.c:265]: Oh, I didn't realize the revision included the version. We probably just need that.

Comment (smcv) [reply src/SDL.c:265]: Yes, `showrev.sh` is careful to include the version number (even if no git info is available) so I think we should be OK with just the revision.

Comment (smcv) [reply src/SDL.c]: OK, it'll now look something like

```
SDL revision: SDL-3.1.3 (Debian 3.1.3+ds-1)
```

or

```
SDL revision: SDL-prerelease-3.1.2-1181-gc5a99f151
```

Comment (slouken) [reply src/SDL.c]: Great.

Comment (slouken) [reply src/SDL.c]: Can we add the SDL revision after the app info?

**Review by slouken — APPROVED** [review]

**Comment by smcv** [comment]
Typical output of `SDL_LOGGING=info ./game-snake` now looks like:

```
App name: Example Snake game
App version: 1.0
App ID: com.example.Snake
SDL revision: SDL-preview-3.1.3-45-g7aab0d38e (smcv's local build)
...
```

where `smcv's local build` is what I set as `SDL_VENDOR_INFO`.

In the Debian packaging, I set `SDL_VENDOR_INFO` to the distro name and the package version, so we get something like `SDL-blahblah (Debian 3.1.3+ds-1)` or `SDL-blahblah (SteamRT 3.1.3+ds-1~steamrt3.1)` to uniquely identify a particular distro package (with enough information to look up the tagged source in a distro-specific location and find out what patches, if any, we're applying).

**Comment by slouken** [comment]
@icculus, can you take a quick look and merge if you like this?

**Comment by icculus** [comment]
Going to add the `value ? value : "<unspecified>"` check to the app name, too. Let me know if there's a reason that doesn't have it.

**Comment by smcv** [comment]
> Going to add the `value ? value : "<unspecified>"` check to the app name, too. Let me know if there's a reason that doesn't have it.

My reason not to have it was that the app name has a non-null default, but I have no problem with programming more defensively.

**Pull request merged**